Makefile

```diff
@@ -190,6 +190,43 @@ docs:
 run-docs:
 	$(MAKE) -C build.assets run-docs
 
+#
+# Remove trailing whitespace in all markdown files under docs/
+#
+.PHONY:docs-fix-whitespace
+docs-fix-whitespace:
+	find docs/ -type f -name '*.md' | xargs sed -i -E 's/\s+$$//g'
+
+#
+# Test docs for trailing whitespace and broken links
+#
+.PHONY:docs-test
+docs-test: docs-test-whitespace docs-test-links
+
+#
+# Check for trailing whitespace in all markdown files under docs/
+#
+.PHONY:docs-test-whitespace
+docs-test-whitespace:
+	if find docs/ -type f -name '*.md' | xargs grep -E '\s+$$'; then \
+		echo "trailing whitespace found in docs/ (see above)"; \
+		echo "run 'make docs-fix-whitespace' to fix it"; \
+		exit 1; \
+	fi
+
+#
+# Run milv in docs to detect broken links.
+# milv is installed if missing.
+#
+.PHONY:docs-test-links
+docs-test-links: DOCS_FOLDERS := $(shell find -name milv.config.yaml | xargs dirname)
+docs-test-links:
+	go get -v github.com/magicmatatjahu/milv
+	for docs_dir in $(DOCS_FOLDERS); do \
```

Comment:
When we introduced Milv we didn't run it on older versions, since now moved out older version and only have current the mass milv fix should be must easier. Once merged, I'll fix the rest of the output.